services/app-web/src/pages/SubmissionSummary/SubmissionSummary.tsx

```diff
@@ -98,25 +105,63 @@ export const SubmissionSummary = (): React.ReactElement => {
         )
     }
 
-    if (error || !submission) return <GenericError />
+    if (error || !submissionAndRevisions) return <GenericError />
 
     const onUnlock = async () => {
-        const result = await unlockMutationWrapper(unlockStateSubmission, submission.id)
+        const result = await unlockMutationWrapper(unlockStateSubmission, submissionAndRevisions.id)
 
         if (result instanceof Error) {
             console.error('ERROR: got an Apollo Client Error attempting to unlock', result)
             setUserVisibleUnlockError('Error attempting to unlock. Please try again.')
         } else if (isGraphQLErrors(result)) {
             console.error('ERROR: got a GraphQL error response', result)
-            setUserVisibleUnlockError('Error attempting to unlock. Please try again.')
+            if (result[0].extensions.code === 'BAD_USER_INPUT') {
+                setUserVisibleUnlockError('Submission is already unlocked. Please refresh and try again.')
+            } else {
+                setUserVisibleUnlockError('Error attempting to unlock. Please try again.')
+            }
         } else {
-            const unlockedSub: DraftSubmission = result
+            const unlockedSub: Submission2 = result
             console.log('Submission Unlocked', unlockedSub)
         }
 
         modalRef.current?.toggleModal(undefined, false)
     }
 
+    const currentRevision = submissionAndRevisions.revisions[0].revision
+
+    const formBinData = Buffer.from(currentRevision.submissionData, 'base64')
+    const submissionResult = toDomain(formBinData)
```

Comment:
_Maybe this is what you are working on now_ - should be in a helper and pushed as close to the client api as possible. No buffers in the JSX please. Looks like we need some kind of library that wraps our api (apollo) functions. Having that data transformation layer separate from the UI  (and with unit tests) is v important.